hal/mbed_ticker_api.c

```diff
@@ -74,15 +74,17 @@ void ticker_insert_event(const ticker_data_t *const data, ticker_event_t *obj, t
         prev = p;
         p = p->next;
     }
+    
+    /* if we're at the end p will be NULL, which is correct */
+    obj->next = p;
```

Comment:
Observe that this is in the generic HAL and us_ticker_set_interrupt is KL25Z specific.  So "if us_ticker_set_interrupt does nothing more than set the interrupt" is only true for KL25Z and only after this change.  And in any case, it's still not a good idea to leave a global resource in a corrupted state before calling an external function.